base/libgit2/repository.jl

```diff
@@ -240,6 +240,18 @@ end
 peel(obj::GitObject) = peel(GitObject, obj)
 
 
+"""
+    peel(T, repo::GitRepo, spec::Union{AbstractGitHash, AbstractString})
+
+Recursively peel the object in `repo` specified by `spec` until an object of type `T` is
+found.
+"""
+function peel{T<:GitObject}(::Type{T}, repo::GitRepo, spec::Union{AbstractGitHash, AbstractString})
```

Comment:
I think Base has mostly made the switch to `where` at this point, so I think this should be
```julia
function peel(::Type{T}, ...) where {T<:GitObject}
```